tests/test_crf.py

```diff
@@ -186,8 +192,10 @@ def test_reduction_none(self):
             denominator = math.log(sum(math.exp(s) for s in all_scores))
             manual_llh.append(numerator - denominator)
 
-        for llh_, manual_llh_ in zip(llh, manual_llh):
-            assert llh_.item() == approx(manual_llh_)
+        if HAS_ASSERT_CLOSE:
+            torch.testing.assert_close(llh, torch.tensor(manual_llh), atol=1e-12, rtol=1e-6)
+        else:
+            torch.testing.assert_allclose(llh, torch.tensor(manual_llh), atol=1e-12, rtol=1e-6)
```

Comment:
Why does `manual_llh` need to be converted to a tensor here but not in the other tests?

Reply:
The `manual_llh` here is a list, and the `llh` is a tensor. The torch assertion function needs the types to be related in a certain way to autoconvert. A list and a tensor do not work, so I need to convert explicitly.